File: packages/sdk-python/examples/06_stream_messages.py
import os
import sys

from bimpeai import BimpeAI

if len(sys.argv) < 3:
    raise SystemExit("usage: 06_stream_messages.py <agent_id> <conversation_id>")

client = BimpeAI(api_key=os.environ.get("BIMPEAI_API_KEY", ""))
for message in client.conversations.messages.stream(sys.argv[1], sys.argv[2]):
    print(f"[{message.role}] {message.message or ''}")

File: examples/typescript/06-stream-messages.ts
// Stream new messages in a conversation in real time. Press Ctrl-C to stop.
import { BimpeAI } from '@bimpeai/sdk';

const client = new BimpeAI({ apiKey: process.env.BIMPEAI_API_KEY ?? '' });
const [agentId, conversationId] = process.argv.slice(2);
if (!agentId || !conversationId) {
  throw new Error('usage: 06-stream-messages.ts <agentId> <conversationId>');
}

const controller = new AbortController();
process.on('SIGINT', () => controller.abort());

for await (const message of client.conversations.messages.stream(agentId, conversationId, {
  signal: controller.signal,
})) {
  console.log(`[${message.role}] ${message.message ?? ''}`);
}
